### PR TITLE
feat: settings migration

### PR DESCRIPTION
**Pull request by cristianoventura**
## Description

This PR adds migrations to the settings schema, ensuring that changes to its structure remain backward-compatible. This allows the settings to work properly even when the schema is updated.

The same pattern will be applied to the Generator schema on a separate PR.

## How to Test

This PR contains one extra commit that is intended to be used for testing purposes and will be reverted before merging.

Before testing this branch, make sure your local settings file (`k6-studio-dev.json`) contains `usageReport.enabled` `false` and `appearance.theme` as `light` so you can validate the changes.

```json
"version": "1.0",
...
"usageReport": { "enabled": false },
"appearance": { "theme": "light" }
```

- Launch k6 studio from this branch
- Check that the theme and usage report are still the same
- Make changes to any other settings in the Settings dialog and check that the previous changes were migrated

Once the settings are saved, the file structure should look like this instead:

```json
"version": "1.0",
...
"general": {
   "usageReport": { "enabled": false },
   "appearance": { "theme": "light" }
}
```

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

Resolves https://github.com/grafana/k6-studio/issues/265

### CODE REVIEW

File: src/components/Settings/AppearanceSettings.tsx
Comment (cristianoventura) [inline src/components/Settings/AppearanceSettings.tsx:18]: This change is intended for testing purposes and will be reverted before merging.

File: src/main.ts
Comment (cristianoventura) [inline src/main.ts:201]: Changes in this file are intended for testing purposes and will be reverted before merging.

Comment (cristianoventura) [inline src/main.ts:201]: Changes in this file are intended for testing purposes and will be reverted before merging.

File: src/schemas/settings/index.ts
Comment (cristianoventura) [inline src/schemas/settings/index.ts:16]: This function will run recursively until the latest version of the schema is reached.

File: src/schemas/settings/v1/index.ts
Comment (cristianoventura) [inline src/schemas/settings/v1/index.ts:114]: Every schema should export a `migrate` function that takes the current version and returns the next.

File: src/types/settings.ts
Comment (e-fisher) [inline src/types/settings.ts:6]: To avoid updating imports every time we introduce a new version we should add export to `schemas/settings/index.ts`
```ts
export * from './v2'
```

**Review by e-fisher — COMMENTED** [review]
LGTM! We'll have another look once test commit is removed

File: src/schemas/settings/v2/index.ts
Comment (cristianoventura) [inline src/schemas/settings/v2/index.ts:19]: This is the same as v1, but I left some breadcrumbs to help us with the first migration when the time comes so no functional change at this time.

**Review by cristianoventura — COMMENTED** [review]
@e-fisher The test commit has been reverted.

**Review by e-fisher — APPROVED** [review]
🙌

**Review by allansson — APPROVED** [review]
LGTM! 👍

**Pull request merged**
